packages/amplify-ui-components/src/components/amplify-verify-contact/amplify-verify-contact.tsx

```diff
@@ -0,0 +1,154 @@
+  tag: 'amplify-verify-contact',
+  shadow: false,
+})
+export class AmplifyVerifyContact {
+  /** Passed from the Authenticator component in order to change Authentication state */
+  @Prop() handleAuthStateChange: AuthStateHandler;
+  /** (Optional) Override default styling */
+  @Prop() overrideStyle: boolean = false;
+  /** Used for the username to be passed to resend code */
+  @Prop() user: CognitoUserInterface;
+  @State() verifyAttr: any;
+
+  errorMessage(err) {
+    if (typeof err === 'string') {
+      return err;
+    }
+
+    return err.message ? err.message : JSON.stringify(err);
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+
+    const form = event.target as HTMLFormElement;
+
+    this.verifyAttr ? this.submit(form.code.value) : this.verify(form.contact.value);
+  }
+
+  submit(code) {
+    const attr = this.verifyAttr;
+
+    if (!Auth || typeof Auth.verifyCurrentUserAttributeSubmit !== 'function') {
+      throw new Error('No Auth module found, please ensure @aws-amplify/auth is imported');
+    }
+
+    Auth.verifyCurrentUserAttributeSubmit(attr, code)
+      .then(data => {
+        logger.debug(data);
+
+        this.handleAuthStateChange(AuthState.SignedIn, this.user);
+        this.verifyAttr = null;
+      })
+      .catch(logger.error);
+  }
+
+  verify(contact: 'email' | 'phone') {
+    if (!contact) {
+      logger.error('Neither Email nor Phone Number selected');
+      return;
+    }
+
+    if (!Auth || typeof Auth.verifyCurrentUserAttribute !== 'function') {
+      throw new Error('No Auth module found, please ensure @aws-amplify/auth is imported');
+    }
+
+    Auth.verifyCurrentUserAttribute(contact)
+      .then(data => {
+        logger.debug(data);
+        this.verifyAttr = contact;
+      })
+      .catch(logger.error);
+  }
+
+  render() {
+    return (
+      <amplify-form-section
+        handleSubmit={event => this.handleSubmit(event)}
+        headerText={VERIFY_CONTACT_HEADER_TEXT}
+        overrideStyle={this.overrideStyle}
+        secondaryFooterContent={
+          <span>
+            <amplify-link onClick={() => this.handleAuthStateChange(AuthState.SignedIn, this.user)}>Skip</amplify-link>
+          </span>
+        }
+        submitButtonText={this.verifyAttr ? VERIFY_CONTACT_SUBMIT_LABEL : VERIFY_CONTACT_VERIFY_LABEL}
+      >
+        {this.verifyAttr ? this.renderSubmit() : this.renderVerify()}
+      </amplify-form-section>
+    );
+  }
+
+  renderSubmit() {
+    return (
+      <div>
+        <amplify-input
+          inputProps={{
+            autocomplete: 'off',
+          }}
+          name="code"
+          overrideStyle={this.overrideStyle}
+          placeholder={CODE_PLACEHOLDER}
+        />
+      </div>
+    );
+  }
+
+  renderVerify() {
+    const user = this.user;
+
+    if (!user) {
+      logger.debug('no user for verify');
+      return null;
+    }
+
+    const { unverified } = user;
+
+    if (!unverified) {
+      logger.debug('no unverified on user');
```

Comment:
This seems slightly strange wording. WDYT about the below? 

```suggestion
      logger.debug('Unverified variable does not exist on user');
```